# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGL1Context.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGL1Context.h
--- a/dom/canvas/WebGL1Context.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGL1Context.h	Wed Jan 08 22:19:14 2020 +0000
@@ -9,25 +9,22 @@
 #include "WebGLContext.h"
 
 namespace mozilla {
+class HostWebGLContext;
 
 class WebGL1Context : public WebGLContext {
  public:
-  static WebGL1Context* Create();
+  static WebGL1Context* Create() { return new WebGL1Context(); }
 
  private:
-  WebGL1Context();
+  WebGL1Context() {}
 
   virtual UniquePtr<webgl::FormatUsageAuthority> CreateFormatUsage(
       gl::GLContext* gl) const override;
 
  public:
-  virtual ~WebGL1Context();
+  virtual ~WebGL1Context(){};
 
   virtual bool IsWebGL2() const override { return false; }
-
-  // nsWrapperCache
-  virtual JSObject* WrapObject(JSContext* cx,
-                               JS::Handle<JSObject*> givenProto) override;
 };
 
 }  // namespace mozilla